hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionReplicaManager.java

```diff
@@ -67,6 +69,11 @@
  */
 public class PartitionReplicaManager implements PartitionReplicaVersionManager {
 
+    // Allow running partition replica sync on generic operation threads? Default is no,
+    // requires explicitly setting system property hazelcast.partition.replica.offload to true
+    private static final String PARTITION_REPLICA_ALLOW_OFFLOAD = "hazelcast.partition.replica.offload";
+    private static final boolean ALLOW_OFFLOAD = Boolean.getBoolean(PARTITION_REPLICA_ALLOW_OFFLOAD);
```

Comment:
This part got me thinking couple of times. As I understand, when `ALLOW_OFFLOAD = true` then we have an offloadable sync requests which lead to differential replication for IMap/ICache on the EE side. At the end, would this parameter be exposed to users and if so what could be a reason not to just set it to true all the time.

Reply:
Differential replica sync is currently off by default because there is an issue that may block cluster from reaching safe state, even though data are correctly replicated. The intent is to fix the issue for GA; then differential partition replica sync will be on by default. However I would still keep the system property switch (not as a `ClusterProperty` advertised in ref manual etc) as a workaround in case an issue is found after GA.

Reply:
I see, thanks